enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,597 @@
+| [Infrastructure API](#infrastructure-api)                  | Add `HighlyAvailableArbiter` as a new value for `ControlPlaneTopology` and `InfrastructureTopology`             |
+| [Installer](#installer-changes)                            | Update install config API to Support arbiter machine types                                                      |
+| [MCO](#mco-changes)                                        | Update validation hook to support arbiter role and add bootstrapping configurations needed for arbiter machines |
+| [Kubernetes](#kubernetes-change)                           | Update allowed `well_known_openshift_labels.go` to include new node role as an upstream carry                   |
+| [ETCD Operator](#etcd-operator-change)                     | Update operator to deploy operands on both `master` and `arbiter` node roles                                    |
+| [library-go](#library-go-change)                           | Update the underlying static pod controller to deploy static pods to `arbiter` node roles                       |
+| [Authentication Operator](#authentication-operator-change) | Update operator to accept minimum 2 kube api servers when `ControlPlaneTopology` is `HighlyAvailableArbiter`    |
+
+### Infrastructure API
+
+We need an authoritative flag to make it easy for components and tools to
+identify if they are running against an `arbiter` backed HA cluster. We are
+proposing adding a new `ControlPlaneTopology` and `InfrastructureTopology` key
+called `HighlyAvailableArbiter` to denote that the cluster is installed with an
+arbiter.
+
+Currently this field will not change, once a cluster is installed as an arbiter
+HA cluster, it will remain that way. There has been discussion on making this
+field mutable to allow a cluster to transition between topologies. However, that
+conversation is beyond the scope of this enhancement, and requires its own
+dedicated enhancement. We will keep this field as static and not support
+topology migrations.
+
+### Installer Changes
+
+We will need to update the installer to have awareness of the explicit intent to
+setup the cluster with an arbiter node. Adding a new machine type similar to the
+flow for `master` and `workers` machines. As an example if the intent is to
+generate 1 `arbiter` machine and 2 `master` machines, the user specifies a
+`MachinePool` `installConfig.arbiterNode` field with the desired replicas.
+
+The `arbiterNode` will be a `MachinePool` object that will enforce at least 1
+replica, and at least 2 replicas for the `controlPlane`. When no `arbiterNode`
+field is supplied regular flow will validate. When `arbiterNode` is supplied 2
+replicas will be valid for for `controlPlane` as long as an `arbiterNode` is
+specified.
+
+`installConfig.yaml`
+
+```yaml
+apiVersion: v1
+baseDomain: devcluster.openshift.com
+compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+arbiterNode:
```

Comment:
A simpler alternative - that would support baremetal clusters only - is to assume that the arbiter is part of the control plane, and identify its existence only by applying the `arbiter` role to one of the hosts in `platform.baremetal.hosts`.
This would result in the _installer_ (rather than a MachineSet) creating the arbiter Node with a taint and a separate pointer ignition.

Reply:
Originally I was leaning that way, but we also did have some testing and cloud deployment desires, that I switched the flow to follow the typical install flow for master, so now it's a full control plane machine at install. I'll update that step by step to correctly reflect that, my mistake on the `MachineSet` wording.